sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -591,20 +590,34 @@ case class FileSourceScanExec(
     logInfo(s"Planning scan with bin packing, max size: $maxSplitBytes bytes, " +
       s"open cost is considered as scanning $openCostInBytes bytes.")
 
+    // Filter files with bucket pruning if possible
+    val filePruning: Path => Boolean = optionalBucketSet match {
+      case Some(bucketSet) =>
+        filePath => bucketSet.get(BucketingUtils.getBucketId(filePath.getName)
+          .getOrElse(sys.error(s"Invalid bucket file $filePath")))
```

Comment:
Could you use `IllegalStateException` instead of `sys.error`?

Reply:
@maropu - I was following [code path](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala#L545) for creating bucketed RDD. Do we want to change this place as well? Just wondering why we prefer `IllegalStateException` here? Is it better error classification?

Reply:
>  I was following code path for creating bucketed RDD. Do we want to change this place as well?

Yea, since the fix is trivial, changing it in this PR looks fine to me.

> Just wondering why we prefer IllegalStateException here? Is it better error classification?

This is not a strict rule, but I think we tend to use `IllegalStateException` for a unexpected code path. For example, see the related previous comment: 
https://github.com/apache/spark/pull/28810#discussion_r439904681

Reply:
+1, `sys.error` crashes the JVM and should be avoided.

Reply:
@maropu , @cloud-fan - sure, updated.